### PR TITLE
first homework

### PR DESCRIPTION
**Pull request by dannykha**
I hope this is good

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
Comment (pdgetrf) [inline .idea/vcs.xml:6]: do not check in .idea folder

File: homework1.iml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      <sourceFolder url="file://$MODULE_DIR$/src/main/java" isTestSource="false" />
+      <sourceFolder url="file://$MODULE_DIR$/src/test/java" isTestSource="true" />
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" name="Maven: junit:junit:4.13" level="project" />
+    <orderEntry type="library" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
+  </component>
+</module>
```
Comment (pdgetrf) [inline homework1.iml:16]: do not check in this iml file

File: src/main/java/Problem1.java [hunk 3]
```diff
@@ -3,6 +3,22 @@
...
+            }
+            else if (target > data[middle]) {
+                left = middle + 1;
+            }
+            else {
+                return middle;
+            }
+        }
+        // commit test
         return -1; // Place holder
```
Comment (pdgetrf) [inline src/main/java/Problem1.java:22]: remove "// place holder" when homework is done.

File: src/main/java/Problem1.java [hunk 4]
```diff
@@ -3,6 +3,22 @@
     // Do not change signature (function name, parameters)
     public static int binarySearch(int[] data, int target) {
         // Add your solution
```
Comment (pdgetrf) [inline src/main/java/Problem1.java:5]: remove this ”// add your solution" when homework is done.

File: src/main/java/Problem2.java [hunk 5]
```diff
@@ -2,6 +2,18 @@
...
     // Do not change signature (function name, parameters)
     public static void bubbleSort(int[] data) {
-        // Add your solution
+        int n = data.length;
+        // ascending data sort!
+
+        for (int i = 0; i < n - 1; i++) {
+            for (int j = 0; j < n - i - 1; j++) {
+                if(data[j] > data[j+1]) {
+                    int temp;
```
Comment (pdgetrf) [inline src/main/java/Problem2.java:11]: combine this line with line 12 into one line
```java
int temp = data[j];
```

File: src/test/java/Problem2Test.java [hunk 6]
```diff
@@ -1,14 +1,29 @@
 import org.junit.Test;
-
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+import java.util.Arrays;
 
 public class Problem2Test {
 
+    private Problem2 sorter = new Problem2();
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:7]: where is this used?

**Review by pdgetrf — APPROVED** [review]
LGTM

File: src/main/java/Problem2.java [hunk 7]
```diff
@@ -2,6 +2,18 @@
 
     // Do not change signature (function name, parameters)
     public static void bubbleSort(int[] data) {
-        // Add your solution
+        int n = data.length;
```
Comment (pdgetrf) [inline src/main/java/Problem2.java:5]: what if the data variable is null?

File: src/test/java/Problem2Test.java [hunk 8]
```diff
@@ -1,14 +1,29 @@
...
+
     // Example test
     // Okay to change
     @Test
     public void testBubbleSort() {
-        assertTrue(false);
+        // assertTrue(false);
         // add your tests here
+        int[] data = {5, 4 , 3, 2, 1};
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:15]: add more test cases. especially the edge case like empty array, null array, array with duplicates.

File: src/test/java/Problem2Test.java [hunk 9]
```diff
@@ -1,14 +1,29 @@
...
 public class Problem2Test {
 
+    private Problem2 sorter = new Problem2();
+
     // Example test
     // Okay to change
     @Test
     public void testBubbleSort() {
-        assertTrue(false);
+        // assertTrue(false);
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:13]: remove code like this that's not used any more.

File: src/test/java/Problem2Test.java [hunk 10]
```diff
@@ -1,14 +1,29 @@
...
         // add your tests here
+        int[] data = {5, 4 , 3, 2, 1};
+        int[] expected = {1, 2, 3, 4, 5};
+        Problem2.bubbleSort(data);
+
+        assertTrue(Arrays.equals(expected, data));
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:19]: use assertArrayEquals()
